src/main/java/org/graylog2/log4j2/GelfAppender.java

```diff
@@ -316,4 +321,8 @@ public static GelfAppender createGelfAppender(@PluginElement("Filter") Filter fi
         return new GelfAppender(name, layout, filter, ignoreExceptions, gelfConfiguration, hostName, includeSource,
                 includeThreadContext, includeStackTrace, additionalFields, includeExceptionCause);
     }
+
+    private static boolean isFQDN(String canonicalHostName) {
+        return canonicalHostName.contains(".") && !canonicalHostName.matches(REGEX_IP_ADDRESS);
```

Comment:
Using a precompiled [`Pattern`](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html) might be preferable.